internal/jobservice/events/client.go

```diff
@@ -0,0 +1,85 @@
+//go:generate moq -out client_moq.go . JobEventReader
+package events
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+// JobEventReader is the interface for retrieving job set event messages
+type JobEventReader interface {
+	GetJobEventMessage(ctx context.Context, jobReq *api.JobSetRequest) (*api.EventStreamMessage, error)
+	Close()
+}
+
+// EventClient is the local struct for retrieving events from the api using the grpc client
+type EventClient struct {
+	config *client.ApiConnectionDetails
+	conn   *grpc.ClientConn
+	mux    *sync.Mutex
+}
+
+// NewEventClient returns a new EventClient
+func NewEventClient(config *client.ApiConnectionDetails) *EventClient {
+	return &EventClient{
+		config: config,
+		mux:    &sync.Mutex{},
+	}
+}
+
+// GetJobEventMessage performs all the steps for obtaining an event message
+func (ec *EventClient) GetJobEventMessage(ctx context.Context, jobReq *api.JobSetRequest) (*api.EventStreamMessage, error) {
+	err := ec.ensureApiConnection()
+	if err != nil {
+		return nil, err
+	}
+	eventClient := api.NewEventClient(ec.conn)
+	log.Debug("looking for job set events")
+	ctx, _ = context.WithTimeout(ctx, time.Duration(5)*time.Second)
```

Comment:
So does this timeout end the stream also?

Reply:
yeah i bet you're right. I'll take another look.

Reply:
removed this timeout for now, we need a gRPC healthcheck endpoint